ush/extractvars_tools.sh

```diff
@@ -0,0 +1,55 @@
+#! /usr/bin/env bash
+
+gen_parmlist() {
+  infileg=$1
+  requestedvar_in_file1=$2
+  varlist=$3
+  file_vars=$(${WGRIB2} "${infileg}")
+  while read -r vari; do
+    if [[ "${file_vars}" =~ ${vari} && -n "${vari}" ]]; then
+      echo "${vari}" >> "${requestedvar_in_file1}"
+    fi
+  done <"${varlist}"
+}
+
+check_atmos() {
+  requestedvar_in_file1=$1
+  requestedvar_in_file2=$2
+  varlistl=$3
+  fnhl=$4
+  requestedvar_in_allgrb2file="${subdata}/requestedvar_in_allgrb2file.txt"
+  rm -rvf "${requestedvar_in_allgrb2file}"
+  cat "${requestedvar_in_file1}" "${requestedvar_in_file2}" >> "${requestedvar_in_allgrb2file}"
+  mapfile -t requestedvar_in_allgrb2file_arr < "${requestedvar_in_allgrb2file}"
+  while read -r vari; do
+    if [[ ! ${requestedvar_in_allgrb2file_arr[*]} =~ ${vari} ]] ;then
+      echo "WARNING: PARM VARIABLE (${vari}) is not available in pgrb and pgrb2b for f${fnhl}."
+    fi
+  done <"${varlistl}"
+}
+
+daily_avg_atmos() {
+  outfile_p=$1
+  fnd=$(printf "%2.2d" "${dcnt}")
```

Comment:
what is `dcnt`?
Please don't use global variables except reserved one such as `RUN`, `cyc`, 
Everything else should be passed to the function.